ext/opentelemetry-ext-jinja2/tests/test_jinja2.py

```diff
@@ -0,0 +1,114 @@
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 2)
+
+        self.assertEqual(spans[0].name, "jinja2.compile")
+        self.assertEqual(spans[0].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[0].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+        self.assertEqual(spans[1].name, "jinja2.render")
+        self.assertEqual(spans[1].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[1].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+    def test_generate_inline_template(self):
+        template = jinja2.environment.Template("Hello {{name}}!")
+        self.assertEqual(
+            "".join(template.generate(name="Jinja")), "Hello Jinja!"
+        )
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 2)
+
+        self.assertEqual(spans[0].name, "jinja2.compile")
+        self.assertEqual(spans[0].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[0].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+        self.assertEqual(spans[1].name, "jinja2.render")
+        self.assertEqual(spans[1].kind, trace_api.SpanKind.INTERNAL)
+        self.assertEqual(
+            spans[1].attributes, {"jinja2.template_name": "<memory>"},
+        )
+
+    def test_file_template(self):
+        loader = jinja2.loaders.FileSystemLoader(TMPL_DIR)
+        env = jinja2.Environment(loader=loader)
+        template = env.get_template("template.html")
+        self.assertEqual(
+            template.render(name="Jinja"), "Message: Hello Jinja!"
+        )
+
+        spans = self.memory_exporter.get_finished_spans()
+        self.assertEqual(len(spans), 5)
+
+        self.assertEqual(spans[0].name, "jinja2.compile")
+        self.assertEqual(spans[1].name, "jinja2.load")
+        self.assertEqual(spans[2].name, "jinja2.compile")
+        self.assertEqual(spans[3].name, "jinja2.load")
+        self.assertEqual(spans[4].name, "jinja2.render")
+
+        self.assertEqual(
+            spans[0].attributes, {"jinja2.template_name": "template.html"},
+        )
+        self.assertEqual(
+            spans[1].attributes,
+            {
+                "jinja2.template_name": "template.html",
+                "jinja2.template_path": os.path.join(
+                    TMPL_DIR, "template.html"
+                ),
+            },
+        )
+        self.assertEqual(
+            spans[2].attributes, {"jinja2.template_name": "base.html"},
+        )
+        self.assertEqual(
+            spans[3].attributes,
+            {
+                "jinja2.template_name": "base.html",
+                "jinja2.template_path": os.path.join(TMPL_DIR, "base.html"),
+            },
+        )
+        self.assertEqual(
+            spans[4].attributes, {"jinja2.template_name": "template.html"},
+        )
```

Comment:
I think a test to check that `uninstrument()` works is missing. You could try to perform an operation, check if spans are generated, then call `uninstrument`, perform another operation and check that no spans are generated.